React/Base/RCTUtils.m

```diff
@@ -337,6 +337,42 @@ BOOL RCTRunningInTestEnvironment(void)
   return isTestEnvironment;
 }
 
+BOOL RCTRunningInAppExtension(void)
+{
+  return [[[[NSBundle mainBundle] bundlePath] pathExtension] isEqualToString:@"appex"];
+}
+
+id RCTSharedApplication(void)
+{
+  if (RCTRunningInAppExtension()) {
+    return nil;
+  }
+  
+  return [[UIApplication class] performSelector:@selector(sharedApplication)];
```

Comment:
Just wondering why it's necessary in this case but not in the case of UIAlertView for example. This code is probably fine, but ARC and performSelector don't always work well together.